spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -588,7 +592,7 @@ protected String mode() {
 
     @Override
     public void doCommit(long epochId, WriterCommitMessage[] messages) {
-      OverwriteFiles overwriteFiles = table.newOverwrite();
+      OverwriteFiles overwriteFiles = table.newOverwrite().toBranch(branch);
```

Comment:
Why do we have to add this call explicitly if we still call `commitOperation` under the hood?